web/static/js/reducers/questionnaire.js

```diff
@@ -136,8 +151,9 @@ const deleteStep = (state, action) => {
   return filter(state, s => s.id != action.stepId)
 }
 
-const changeStep = (state, stepId, func: (step: Step) => Step) => {
+function changeStep<T: Step>(state, stepId, func: (step: Object) => T) {
```

Comment:
These type annotations make Flow happy with a polymorphic `func` parameter. I suggest you not to mess with it :P